lib/commands/gesture.js

```diff
@@ -114,13 +114,108 @@ helpers.handleTap = async function (gesture) {
   return await this.uiAutoClient.sendCommand(cmd);
 };
 
+helpers.isDrag = function (gestures) {
+  return (
+      gestures.length === 4 &&
+      gestures[0].action === 'press' &&
+      gestures[1].action === 'wait' &&
+      gestures[2].action === 'moveTo' &&
+      gestures[3].action === 'release'
+  );
+};
+
+helpers.getCoordinates = async function(gesture) {
+  let el = gesture.options.element;
+
+  // defaults
+  let coordinates = {x: 0, y: 0, areOffsets: false};
+
+  // figure out the element coordinates.
+  if (el) {
+    let command = `au.getElement('${el}').rect()`;
+    let rect = await this.uiAutoClient.sendCommand(command);
+    let pos = {x: rect.origin.x, y: rect.origin.y};
+    let size = {w: rect.size.width, h: rect.size.height};
+
+    // defaults
+    let offsetX = 0;
+    let offsetY = 0;
+
+    // get the real offsets
+    if (gesture.options.x || gesture.options.y) {
+      offsetX = (gesture.options.x || 0);
+      offsetY = (gesture.options.y || 0);
+    } else {
+      offsetX = (size.w / 2);
+      offsetY = (size.h / 2);
+    }
+
+    // apply the offsets
+    coordinates.x = pos.x + offsetX;
+    coordinates.y = pos.y + offsetY;
+  } else {
+    // moveTo coordinates are passed in as offsets
+    coordinates.areOffsets = (gesture.action === 'moveTo');
+    coordinates.x = (gesture.options.x || 0);
+    coordinates.y = (gesture.options.y || 0);
+  }
+  return coordinates;
+};
+
+helpers.validateGestureType = function(gesture, expected) {
+  if (gesture.action !== expected) {
+    throw new errors.UnknownError(
+      `'${expected}' gesture expected, but ${gesture.action} was passed instead!`);
+  }
+};
+
+helpers.applyMoveToOffset = function(firstCoordinates, secondCoordinates) {
+  if (secondCoordinates.areOffsets) {
+    return {
+      x: firstCoordinates.x + secondCoordinates.x,
+      y: firstCoordinates.y + secondCoordinates.y,
+    };
+  } else {
+    return secondCoordinates;
+  }
+};
+
+helpers.handleDrag = async function (gestures) {
+  // get gestures
+  let press = gestures[0];
+  this.validateGestureType(press, 'press');
+  let wait = gestures[1];
+  this.validateGestureType(wait, 'wait');
+  let moveTo = gestures[2];
+  this.validateGestureType(moveTo, 'moveTo');
+
+  // get drag data
+  let pressCoordinates = await this.getCoordinates(press);
+  let duration = (parseInt(wait.options.ms) / 1000);
```

Comment:
`parseInt` needs a second parameter, `10`, otherwise the behavior is unpredictable.

```js
let duration = parseInt(wait.options.ms, 10) / 1000;
```